src/extension/content-script/batteries/StackOverflow.ts

```diff
@@ -1,41 +1,102 @@
+import { Battery } from "~/types";
+
 import getOriginData from "../originData";
 import setLightningData from "../setLightningData";
 
 const urlMatcher =
-  /^https:\/\/stackoverflow\.com\/users\/(\d+)\/(\w+)(\?tab=(\w+))?.*/;
+  /^https:\/\/stackoverflow\.com\/(users|questions)\/(\d+)\/(\w+).*/;
+
+const battery = async (): Promise<void> => {
+  const urlParts = document.location.pathname.split("/");
+  const route = urlParts[1];
+  const userOrQuestionId = urlParts[2];
+  let lightningData = null;
 
-const battery = (): void => {
-  const urlParams = new URLSearchParams(window.location.search);
-  const tab = urlParams.get("tab");
+  if (route === "users") {
+    lightningData = await handleProfilePage(userOrQuestionId);
+  }
 
-  if (tab === "profile") {
-    handleProfilePage();
+  if (route === "questions") {
+    lightningData = await handleQuestionPage(userOrQuestionId);
   }
+
+  if (lightningData) setLightningData([lightningData]);
 };
 
-function handleProfilePage() {
-  const aboutElement = document.querySelector<HTMLElement>(
-    ".js-about-me-content"
-  );
-
-  const text = aboutElement?.innerText as string;
-  const match = text.match(/(⚡:?|lightning:|lnurl:)\s?(\S+@\S+)/i);
-
-  if (match) {
-    setLightningData([
-      {
-        method: "lnurl",
-        recipient: match[2],
-        ...getOriginData(),
-        description: aboutElement?.innerText ?? "",
-        icon:
-          document.querySelector<HTMLImageElement>(
-            `.js-usermini-avatar-container img`
-          )?.src ?? "",
-        name: document.title,
-      },
-    ]);
+function findLightningAddressInText(text: string) {
+  const match = text.match(/(⚡:?|lightning:|lnurl:)\s?(\S+@[\w-.]+)/i);
+  if (match) return match[2];
+}
+
+function htmlToText(html: string) {
+  const temp = document.createElement("div");
+  temp.innerHTML = html;
+  return temp.textContent;
+}
+
+async function handleQuestionPage(questionId: string): Promise<Battery | null> {
+  const questionInfo = await fetch(
+    // The filter can be generated here: https://api.stackexchange.com/docs/users-by-ids
+    `https://api.stackexchange.com/2.2/questions/${questionId}?site=stackoverflow&filter=!9bOY8fLl6`
+  ).then((res) => res.json());
+  if (questionInfo.error_id) {
+    return null;
+  }
+
+  const questionData = questionInfo.items[0];
+
+  if (!questionData || !questionData.accepted_answer_id) {
+    return null;
+  }
+
+  const answerInfo = await fetch(
```

Comment:
We prefer Axios over fetch but that's not blocking: https://github.com/getAlby/lightning-browser-extension#code-format--preferences

Reply:
Done!